src/main/java/duke/logic/Parser.java

```diff
@@ -0,0 +1,275 @@
+                }
+                //strip leading get rid of white spaces infront, back alr dealt with
+                time = time.stripLeading();
+                return new EventCommand(taskList, description, Parser.parseTime(time));
+            }
+        } catch (IllegalDescriptionException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! The description cannot be empty.");
+            };
+        } catch (IllegalTimeException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! The time specified is either invalid or empty.");
+            };
+        } catch (IllegalTokenException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! You are mising the \"/by\" or \"/at\" token.");
+            };
+        }
+        return () -> {
+            System.out.println(" ☹ OOPS!!! I'm sorry, but I don't know what that means. :-(");
+        };
+    }
+    
+    /**
+     * Parses string to the appropriate time it represents, if any.
+     *
+     * @param str the string to parse.
+     * @return the LocalDate object representing time.
+     * @throws IllegalTimeException If unable to parse the string to a valid time.
+     */
+    private static LocalDate parseTime(String str) throws IllegalTimeException {
+        String sDay;
+        String sMonth;
+        String sYear;
+        //ddmmyyyy
+        String[] arr = new String[3];
+        //forward slash, hyphen, dot, or whitespace, otherwise no separator cases
+        if (str.indexOf('/') >= 0) {
+            arr = Parser.parseTimeHelper('/', str);
+        } else if (str.indexOf('-') >= 0) {
+            arr = Parser.parseTimeHelper('-', str);
+        } else if (str.indexOf('.') >= 0) {
+            arr = Parser.parseTimeHelper('.', str);
+        } else if (str.indexOf(' ') >= 0) {
+            arr = Parser.parseTimeHelper(' ', str);
+        } else if (str.length() == 4 || str.length() == 6 || str.length() == 8) {
+            arr[0] = str.substring(0, 2);
+            arr[1] = str.substring(2, 4);
+            if (str.length() > 4) {
+                arr[2] = str.substring(4);
+            }
+        }   //if any of these if statements go through, arr will have atleast 2 no null elements
+        if (arr[0] == null || arr[1] == null) { //no method of parsing, arr[1] check redundant but just in case
+            throw new IllegalTimeException("Unable to parse due to invalid format.");
+        }
+        sDay = arr[0].strip();
+        sMonth = arr[1].strip();
+        //check if d/dd and m/mm/MMM
+        if (sDay.length() < 1 || sDay.length() > 2 || sMonth.length() < 1 || sMonth.length() > 3) {
+            throw new IllegalTimeException("Invalid day or month values.");
+        }
+        //convert month to number
+        if (sMonth.length() == 3) {  //length of 3 may indicate abbrev
+            sMonth = Parser.months.get(sMonth.toLowerCase());
```

Comment:
I like that you checked the month manually, but maybe you could use the LocalDate's native parsing function to simplify the code a bit.